large-message-core/src/main/java/com/bakdata/kafka/AbstractLargeMessageConfig.java

```diff
@@ -301,12 +312,33 @@ private BlobStorageClient createAmazonS3Client() {
         this.getAmazonEndpointOverride().ifPresent(clientBuilder::endpointOverride);
         this.getAmazonRegion().ifPresent(clientBuilder::region);
         this.getAmazonCredentialsProvider().ifPresent(clientBuilder::credentialsProvider);
+        if (this.getTlsTrustManagerProvider().isPresent()) {
+            clientBuilder.httpClientBuilder(
+                    ApacheHttpClient.builder().tlsTrustManagersProvider(
```

Comment:
Which client is used by default? Would it be instead feasible to specify a HttpClientProvider and the user can implement it? Then we can get rid of the dependency on the apache module

Reply:
Thanks for the comments, I will try to figure out if it is feasible

Reply:
The default is software.amazon.awssdk.core.internal.http.loader.DefaultSdkHttpClientBuilder
And I update the PR to let user specify the HttpClientProvider instead of TlsTrustManagerProvider